pallets/messages/src/lib.rs

```diff
@@ -410,7 +410,17 @@ impl<T: Config> Pallet<T> {
 		// Decode SCALE encoded CID into string slice
 		let cstr: &str = sp_std::str::from_utf8(&in_cid[..]).map_err(|_| Error::<T>::InvalidCid)?;
 		ensure!(cstr.len() > 2, Error::<T>::InvalidCid);
-		ensure!(&cstr[0..2] != "Qm", Error::<T>::UnsupportedCidVersion);
+
```

Comment:
Are you ok with renaming cstr to cid_str? 

We could also simplify the code using starts_with.

```
  ensure!(!cid_str.starts_with("Qm"), Error::<T>::UnsupportedCidVersion);
```